### PR TITLE
Support kubernetes v1.24

### PR DESCRIPTION
**Pull request by d-kuro**
Support Kubernetes v1.24.

* update deps
* update tools
* update Go version to v1.18

### CODE REVIEW

File: api/v1beta1/conversion_test.go [hunk 1]
```diff
@@ -18,75 +20,83 @@ func TestCompatibility(t *testing.T) {
...
+		}
+		if err := scheme.Convert(&tmp1, &v1beta2Cluster, nil); err != nil {
+			t.Fatal(err)
+		}
+		if err := scheme.Convert(&v1beta2Cluster, &tmp2, nil); err != nil {
+			t.Fatal(err)
+		}
+		if err := scheme.Convert(&tmp2, &v1beta1Cluster2, nil); err != nil {
+			t.Fatal(err)
+		}
```
Comment (masa213f) [inline api/v1beta1/conversion_test.go:46]: Could you tell me why we converted it four times? Sorry, I forgot :(

Comment (d-kuro) [reply api/v1beta1/conversion_test.go:46]: Certainly no need for 4 conversions...

When I implemented conversion in the past, I tried to prepare internal objects in addition to v1beta1 and v1beta2.
In the controller logic, using an internal objcet has the advantage of not having to change the api version used by the controller when the api version increases in the future.

I prepare an internal object, you will need to do the conversion 4 times:

* v1beta1 -> internal -> v1beta2 -> internal -> v1beta1

However, the implementation became so complex that we decided not to adopt internal objects.
I had forgotten to make the fix to the test at that time.

Comment (d-kuro) [reply api/v1beta1/conversion_test.go:46]: I will modify it to the proper conversion logic.
Thanks for review 🙏

Comment (d-kuro) [reply api/v1beta1/conversion_test.go]: fixed: https://github.com/cybozu-go/moco/pull/429/commits/27856251cf9b6b066f98a58853b213e0d9e68767

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**